src/library/analysisfeature.cpp

```diff
@@ -13,21 +13,25 @@
 #include "library/dlganalysis.h"
 #include "widget/wlibrary.h"
 #include "controllers/keyboard/keyboardeventfilter.h"
-#include "analyzer/analyzerqueue.h"
 #include "sources/soundsourceproxy.h"
 #include "util/dnd.h"
 #include "util/debug.h"
 
 const QString AnalysisFeature::m_sAnalysisViewName = QString("Analysis");
 
-AnalysisFeature::AnalysisFeature(Library* parent,
-                               UserSettingsPointer pConfig,
-                               TrackCollection* pTrackCollection) :
+namespace {
+
+// Utilize all available cores for batch analysis of tracks
+const int kNumberOfAnalyzerThreads = math_max(1, QThread::idealThreadCount());
```

Comment:
As I told you in the comment, it is a good idea to offer the user the possibility to limit the max amount of threads used for analysis.  In my implementation, there's a dropdown list from 1 to idealThreadCount() and it is initialized with idealThreadCount().

Reply:
I don't remember that there is such an option in any of the competing products? The common user even doesn't know what a "thread" is. I suppose that >95% will just go with the default settings. Should we really provide such an option or just think about a sensible default (either fixed constant or load-based heuristic) which will satisfy 98% of our users? Each option comes at a price including tests, maintenance, and documentation in the manual.